freqtrade/rpc/api_server/ws/serializer.py

```diff
@@ -0,0 +1,65 @@
+import json
+import logging
+from abc import ABC, abstractmethod
+
+import msgpack
+import orjson
+
+from freqtrade.rpc.api_server.ws.proxy import WebSocketProxy
+
+
+logger = logging.getLogger(__name__)
+
+
+class WebSocketSerializer(ABC):
+    def __init__(self, websocket: WebSocketProxy):
+        self._websocket: WebSocketProxy = websocket
+
+    @abstractmethod
+    def _serialize(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def _deserialize(self, data):
+        raise NotImplementedError()
+
+    async def send(self, data: bytes):
+        await self._websocket.send(self._serialize(data))
+
+    async def recv(self) -> bytes:
+        data = await self._websocket.recv()
+
+        return self._deserialize(data)
+
+    async def close(self, code: int = 1000):
+        await self._websocket.close(code)
+
+# Going to explore using MsgPack as the serialization,
+# as that might be the best method for sending pandas
+# dataframes over the wire
+
+
+class JSONWebSocketSerializer(WebSocketSerializer):
+    def _serialize(self, data):
+        return json.dumps(data)
+
+    def _deserialize(self, data):
+        return json.loads(data)
```

Comment:
you may wan to use orjson for this instead (basically dropping this).

fastapi uses it (so we depend on it already) - it has clear performance advantages - and is a bit safer when it comes to transferring dataframes.
As it's already available - there's no point in using the builtin json serializer.